jwe_parse.go

```diff
@@ -0,0 +1,60 @@
+package jwe
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+func ParseEncrypted(input string) (*jwe, error) {
+
+	if strings.HasPrefix(input, "{") {
+		return nil, errors.New("don't support full JWE")
+	}
+
+	return parseEncryptedCompact(input)
+}
+
+func parseEncryptedCompact(input string) (*jwe, error) {
+	parts := strings.Split(input, ".")
+
+	if len(parts) != 5 {
+		return nil, errors.New("encrypted token contains an invalid number of segments")
+	}
+
+	jwe := &jwe{}
```

Comment:
I guess we should avoid constructing this directly, if there is a `NewJWE` function, even though it is internal?

Reply:
The `NewJWE` creates a new JWE with key and plaintext. Here is a JWE in compact form, we don't know key.
We work with final state and should parse it by parts.